docs/guides/user-containers/readme.md

```diff
@@ -6,7 +6,55 @@ applications to be run on Rabbit nodes with access to NNF ephemeral and persiste
 !!! note
 
     The following is a limited look at User Containers.  More content will be
-    provided after the RFC has been finalized.
+    provided after the RFC has been finalized. This is a work in progress.
+
+## Environment Variables
+
+Two sets of environment variables are available with container workflows: Container and Compute Node. The former are the variables that are available inside the user containers. The latter are the variables that are provided back to the DWS workflow, which in turn are collected by Flux and provided to compute nodes. See the Flux documentation for more details.
+
+### Container Environment Variables
+
+These variables are provided for use inside the container. They can be used as part of the
+container command in the NNF Container Profile or within the container itself.
+
+#### Container Hostnames and Domains
+
+Containers can contact one another via Kubernetes cluster networking. This functionality is provided by DNS. Environment variables are provided that allow a user to be able to piece together the FQDN so that the other containers can be contacted.
+
+This example demonstrates an MPI container workflow, with two worker nodes. Two worker nodes means two containers running on two NNF nodes.
+
+```console
+mpiuser@my-container-workflow-launcher:~$ env | grep NNF
+NNF_CONTAINER_HOSTNAMES=my-container-workflow-launcher my-container-workflow-worker-0 my-container-workflow-worker-1
+NNF_CONTAINER_DOMAIN=default.svc.cluster.local
+NNF_CONTAINER_SUBDOMAIN=my-container-workflow-worker
+```
+
+The container FQDN consists of the following: `<HOSTNAME>.<SUBDOMAIN>.<DOMAIN>`. To contact the other worker container from worker 0, `my-container-workflow-worker-1.my-container-workflow-worker.default.svc.cluster.local` would be used.
+
+For MPI-based containers, an alternate way to retrieve this information is to look at the default `hostfile`, provided by `mpi-operator`. This file lists out all the worker nodes' FQDNs:
+
+```console
+mpiuser@my-container-workflow-launcher:~$ cat /etc/mpi/hostfile
+my-container-workflow-worker-0.my-container-workflow-worker.default.svc slots=1
+my-container-workflow-worker-1.my-container-workflow-worker.default.svc slots=1
+```
+
+### Compute Node Environment Variables
+
+These environment variables are provided to the compute node via Flux by way of the DWS Workflow. Note that these environment variables are consistent across all the compute nodes for a given workflow. At this time, there is no way to provided compute specific environment variables.
```

Comment:
Do you ever intend to provide compute specific environment variables? If not or at least not likely, consider removing "At this time,"

Reply:
I don't believe Flux has a way to do this, unless we get specific with the envionrment variable names themselves. I'll remove that part.